### DISCUSSION TITLE
WasmEdge + WASI-NN OpenVINO Backend

### DISCUSSION THREAD

**Discussion by omkarprabhu-98** [WasmEdge/WasmEdge#2381] [Q&A]
I followed [this guide ](https://wasmedge.org/book/en/contribute/build_from_src/plugin_wasi_nn.html) to build wasmedge with openvino support

On running wasmedge on a fresh bash session I get the following message: 

<img width="1255" alt="Untitled" src="https://user-images.githubusercontent.com/23053768/227746199-be8a3096-5288-49d2-973f-f535aa8984df.png">

Although this goes away when I `source /opt/intel/openvino_2022/setupvars.sh` I was hoping to not do this everytime I run wasmedge

Did I miss something in the build process?

Thanks!

**Comment by hydai** [comment]
Since there are several ways to install NN dependencies, we do not source the environment file for users. You may need to source it on a new session or add it to the shell script.

**Comment by omkarprabhu-98** [comment]
Thanks!

Actually I'm using crun built with wasmedge support and when crun calls wasmedge (to load the wasm module which runs a openvino model) an error is thrown because of not loading the library 

Would you happen to know where I could potentially source this files so this works properly? Is there a way to statically link this openvino library when building wasmedge?
